## Capítulo 1.1.: Modelos estadísticos para control de calidad

### Caso 1.1.1: Análisis de capacidad de proceso

In [2]:
import numpy as np
import math
import scipy.stats as st

#### Ejemplo 5:

Un ventilador mecánico o respirador es una máquina, de accionamiento eléctrico, -controlado mediante procesador- electromagnético o neumático, que suple la ventilación pulmonar espontánea por una ventilación mecánica en personas con una insuficiencia respiratoria, ya sea aguda o crónica. El gas de respiración suele estar enriquecido con oxígeno.

Una reconocida empresa de bioingeniería se encuentra en la fase terminal del diseño de su nuevo respirador artificial para utilizar en pacientes con graves deficiencias de respiración. El aparato tiene como objeto brindar hasta 95% de oxígeno puro a través de su depósito de aire, el cuál es comprimido neumáticamente varias veces para proporcionar al paciente aire circundante o, en la mayoría de los casos, una mezcla de aire y oxígeno.

La empresa ha producido un total de 50 unidades de prueba y, según los resultados que arroja el medidor utilizado, se tiene que la cantidad máxima promedio de oxígeno que suministra el artefacto es de 93.4% con una desviación estándar de 1.5%. Los inversores del proyecto, además de requerir que la máquina suministre un máximo promedio de 95% de oxígeno también solicitan que la discrepancia con dicho valor sea a lo máximo 2% más bajo para dar marcha al proyecto a gran escala (suponga que los datos siguen una distribución normal).

**a) Estimar la Capacidad Potencial (Cp):**

R/ Se sabe que el índice de capacidad del proceso Cp esta dado por:

$$\rightarrow C_{p}=\frac{T}{6\sigma'}$$

Y:
$$\rightarrow T = 2\nabla = LSE - LIE$$

Donde:
* $\nabla$ es la discrepancia y de ahora en adelante se le asignará la letra 'D', 
* $\sigma'$ que es la desviación estándar se le asignará la letra 's',y 
* 'T' es la tolerancia. 

A continuación se procederá determinar la capacidad del proceso:

In [3]:
D = 0.02
LIE = 0.95 - D
LSE = 0.95 + D
T = 2*D
s = 0.015 # Se toma s como el mejor estimador de sigma.

Cp = T/(6*s)
print(Cp)

0.4444444444444445


Dado que $C_{p} < 0.67$ entonces esto por lo pronto indica que el proceso tiene una muy pobre capacidad de fabricar producto dentro de las especificaciones, y según los criterios se trata de un proceso no adecuado para el trabajo, por lo que requiere modificaciones serias.

**b) Estimar la Capacidad Real (Cpk):**

R/ Cuando el proceso no está centrado, se emplea el **índice de capacidad unilateral** o **índice de capacidad real (Cpk)**:

$$\rightarrow C_{pk}=min(C_{pki}, C_{pks})$$

$$\rightarrow C_{pk}=min(C_{pki}=\frac{\bar{X'}-LIE}{3\sigma'}, C_{pks}=\frac{LSE-\bar{X'}}{3\sigma'})$$

Donde:

* $\bar{X'}$ es la media del proceso, y
* $\sigma'$ es la desviación del proceso.

Por tanto, el índice Cpk es:

In [4]:
X = 0.934 # Se toma X' como el mejor estimador de mu

Cpki = np.round((X - LIE)/(3*s),2)
Cpks = np.round((LSE - X)/(3*s),2)

Cpk = np.min([Cpki, Cpks])

print("Cpki = "+ str(Cpki))
print("Cpks = "+ str(Cpks))
print("Cpk = "+ str(Cpk))

Cpki = 0.09
Cpks = 0.8
Cpk = 0.09


Dado que Cpk $\approx$ 0.09, esto confirma que el proceso no es realmente capaz y por lo menos tiende a no cumplir con una de las especificaciones de calidad. Dicho lo anterior, tal parece que no se está cumpliendo en mayor medida con el límite inferior de calidad, pues corresponde a Cpki $\approx$ 0.09. De igual forma, el proceso tampoco parece cumplir con el límite superior de especificación, ya que Cpks $\approx$ 0.8.

**c) Calcule los límites de fluctuación natural del proceso (LFNP):**

R/ Se tiene que los límites de fluctuación natural del proceso están dados por la siguiente fórmula:

$$\rightarrow LFN_{i,s} = \bar{X'} \pm 3\sigma'$$

Donde:

* $\bar{X'}$ es la media del proceso, y
* $\sigma'$ es la desviación estándar del proceso.

Entonces, los límites de fluctuación son:

In [5]:
LIFN = np.round(X - 3*s,4)
LSFN = np.round(X + 3*s,4)

print("LIFN = " + str(LIFN) + " y LSFN = " + str(LSFN))

LIFN = 0.889 y LSFN = 0.979


Se compara si LIFN es mayor o igual a LIE y si LSFN es menor o igual a LSE:

In [6]:
print(LIFN>=LIE, LSFN<=LSE)

False False


Se confirma que ninguno de los límites de fluctuación natural del proceso se encuentran dentro de las especificaciones de calidad.

**d) Dado que para obtener la financiación es prioritario que las máquinas cumplan con los requerimientos mínimos de calidad y las máquinas que suministran mayor cantidad de oxígeno se pueden calibrar para optimizar el suministro. Por ahora a los socios solo les interesa saber ¿Qué porcentaje de respiradores artificiales no cumplen con el límite inferior de especificación?**

R/ Inicialmente se debe determinar el valor Z estandarizado para dicho límite (LIE). Esto se obtiene con la siguiente fórmula:

$$\rightarrow Z_{inf}=\frac{LIE-\bar{X'}}{\sigma'}$$

Obteniendo el siguiente resultado:

In [7]:
Zinf = np.round((LIE - X)/s,4)
print(Zinf)

-0.2667


In [8]:
P = np.round(st.norm.cdf(Zinf),4)
print(P)

0.3949


Es decir que la proporción de producto no conforme generado por debajo del límite inferior de especificación es del 39.49%. El porcentaje es alto, representarían aproximadamente 3.949 unidades defectuosas en cada lote de 10.000 unidades, pero se podría mejorar reduciendo un poco más la variabilidad del proceso.

**e) Suponiendo que se pudiera reducir drásticamente la variabilidad del proceso, ¿Qué valor máximo debería tener la desviación estándar para lograr a lo sumo 5% de unidades defectuosas por debajo del LIE?**

Inicialmente se debe despejar $\sigma'$ de la fórmula de Z estandarizado inferior, así:

$$Z_{inf}=\frac{LIE-\bar{X'}}{\sigma'}\rightarrow\sigma'_{inf}=\frac{LIE-\bar{X'}}{Z_{inf}}$$

Donde:
* $Z_{inf}$ es el valor de Z estandarizado inferior que representa una probabilidad de 5.0% de obtener producto no conforme por debajo de LIE.

Por lo tanto:

In [11]:
VO_inf = 0.05
Zinf_vo = np.round(st.norm.ppf(VO_inf),4)

print("Zinf = "+str(Zinf_vo))

Zinf = -1.6449


Entonces, los valores entre los cuales puede oscilar $\sigma'$ para obtener a lo sumo 5.0% de producto no conforme son:

In [13]:
s_inf_vo = np.round((LIE - X)/Zinf_vo,5) # Cuando el proceso entrega 5% de producto no conforme por debajo del límite inferior de especificación.
print("Sinf = " + str(s_inf_vo))

Sinf = 0.00243


Por tanto, el proceso debe tener mejorar su precisión debido a que la desviación estándar debe ser máximo 0.243% para obtener a lo sumo 5% de no conformidades.